Zimbra/zm-mailbox
# ZCS-3594 : Create API to trigger contact backup thread

1. Created SOAP API to trigger contact backup.
2. API with options to start/stop/status contact backup.
3. API for admin users only.
4. Updated soap-admin.txt with API.

Testing done :
1. Manually tested api
2. Added unit tests to verify api

Testing to be done by QA :
1. Verify api behavior
2. Verify contact backup functionality

API :
```
<ContactBackupRequest [task="start|stop|status"]>
    [<mailboxIds>
        <id>xxx</id>
        <id>xxx</id>
        <id>xxx</id>
        ...
    </mailboxIds>
    ]
</ContactBackupRequest>
```

task : It's an optional attribute. If not mentioned, "start" considered by default.
start : start contact backup
stop : stop contact backup
status : status of currently running contact backup
mailboxIds : optional wrapper element for ids. if not mentioned all the mailboxes on the server will be processed.
ids : mailbox ids which will be processed.

```
<ContactBackupResponse>
    [<doneIds>
        <id>xxx</id>
        <id>xxx</id>
        <id>xxx</id>
        ...
    </doneIds>
    ]
    [<skippedIds>
        <id>xxx</id>
        <id>xxx</id>
        <id>xxx</id>
        ...
    </skippedIdss>
    ]
</ContactBackupResponse>

```
doneIds : wrapper for all ids which are processed. received in case of "status/stop" request only.
skippedIds : wrapper for all mailbox ids which are skipped. received in case of "status/stop".
id : mailbox ids
